### PR TITLE
CLOUDP-224541: Added terminationProtection flag

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:
Added terminationProtection flag

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by s-urbaniak — APPROVED** [review]
Please add e2e tests as a follow-up, preferably before we start the release tomorrow. If not, that's ok as well.

**Comment by s-urbaniak** [comment]
@igor-karpukhin ^

**Review by josvazg — APPROVED** [review]

**Comment by s-urbaniak** [comment]
@igor-karpukhin instead of commenting I took the freedom to put a commit on top to untangle the nested ifs, PTAL

File: pkg/controller/atlasdeployment/atlasdeployment_controller.go [hunk 1]
```diff
@@ -274,44 +274,45 @@ func (r *AtlasDeploymentReconciler) handleDeletion(
...
-	return false, workflow.OK()
+
+	if err := r.cleanupBindings(workflowCtx.Context, deployment); err != nil {
+		result := workflow.Terminate(workflow.Internal, err.Error())
+		log.Errorw("failed to cleanup deployment bindings (backups)", "error", err)
+		return true, result
+	}
+
+	switch {
+	case customresource.IsResourceProtected(deployment, r.ObjectDeletionProtection):
```
Comment (s-urbaniak) [inline pkg/controller/atlasdeployment/atlasdeployment_controller.go:292]: suggestion: rename this to `customresource.IsResourcePolicyKeepOrDefault`. The method is the very same as below but with a default `bool`.

File: pkg/api/v1/atlasdeployment_types_test.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ func init() {
 	excludedClusterFieldsTheirs["replicationFactor"] = true
 
 	// Termination protection
-	excludedClusterFieldsTheirs["terminationProtectionEnabled"] = true
+	// excludedClusterFieldsTheirs["terminationProtectionEnabled"] = true
```
Comment (josvazg) [inline pkg/api/v1/atlasdeployment_types_test.go:36]: why is this needed?

**Review by josvazg — APPROVED** [review]
